pkg/kubelet/dockertools/manager.go

```diff
@@ -502,11 +505,13 @@ func (dm *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
 		_, err := dm.client.InspectImage(image)
 		if err == nil {
 			containerStatus.State.Waiting = &api.ContainerStateWaiting{
-				Reason: fmt.Sprintf("Image: %s is ready, container is creating", image),
+				Reason:  "ContainerCreating",
+				Message: fmt.Sprintf("Image: %s is ready, container is creating", image),
 			}
 		} else if err == docker.ErrNoSuchImage {
 			containerStatus.State.Waiting = &api.ContainerStateWaiting{
-				Reason: fmt.Sprintf("Image: %s is not ready on the node", image),
+				Reason:  "NoSuchImage",
```

Comment:
InspectImage only returns ErrNoSuchImage, and Pullmage does have both status codes: 200 for ErrNoSuchImage and 500 for unresponsive repository, but we didn't cache that result from different goroutine. Let's fix this one in a separate pr.